### PR TITLE
[IR] Add getDataLayout() helpers to BasicBlock and Instruction

### PR DESCRIPTION
**Pull request by nikic**
This is a helper to avoid writing `getModule()->getDataLayout()`. I regularly try to use this method only to remember it doesn't exist...

`getModule()->getDataLayout()` is also a common (the most common?) reason why code has to include the Module.h header.

### CODE REVIEW

File: llvm/include/llvm/IR/BasicBlock.h [hunk 1]
```diff
@@ -218,6 +219,11 @@ class BasicBlock final : public Value, // Basic blocks are data objects also
                             static_cast<const BasicBlock *>(this)->getModule());
   }
 
+  /// Get the data layout of the module this instruction belongs to.
```
Comment (dtcxzyw) [inline llvm/include/llvm/IR/BasicBlock.h:222]:
```suggestion
  /// Get the data layout of the module this basic block belongs to.
```

File: llvm/lib/Transforms/Scalar/LoopLoadElimination.cpp [hunk 2]
```diff
@@ -96,7 +96,7 @@ struct StoreToLoadForwardingCandidate {
     Value *LoadPtr = Load->getPointerOperand();
     Value *StorePtr = Store->getPointerOperand();
     Type *LoadType = getLoadStoreType(Load);
-    auto &DL = Load->getParent()->getModule()->getDataLayout();
+    auto &DL = Load->getParent()->getDataLayout();
```
Comment (dtcxzyw) [inline llvm/lib/Transforms/Scalar/LoopLoadElimination.cpp:99]:
```suggestion
    auto &DL = Load->getDataLayout();
```

File: llvm/lib/Transforms/Scalar/LoopLoadElimination.cpp [hunk 3]
```diff
@@ -225,7 +225,7 @@ class LoadEliminationForLoop {
       // Only propagate if the stored values are bit/pointer castable.
       if (!CastInst::isBitOrNoopPointerCastable(
               getLoadStoreType(Store), getLoadStoreType(Load),
-              Store->getParent()->getModule()->getDataLayout()))
+              Store->getParent()->getDataLayout()))
```
Comment (dtcxzyw) [inline llvm/lib/Transforms/Scalar/LoopLoadElimination.cpp:228]:
```suggestion
              Store->getDataLayout()))
```

File: llvm/lib/Transforms/Scalar/LoopLoadElimination.cpp [hunk 4]
```diff
@@ -459,7 +459,7 @@ class LoadEliminationForLoop {
 
     Type *LoadType = Initial->getType();
     Type *StoreType = Cand.Store->getValueOperand()->getType();
-    auto &DL = Cand.Load->getParent()->getModule()->getDataLayout();
+    auto &DL = Cand.Load->getParent()->getDataLayout();
```
Comment (dtcxzyw) [inline llvm/lib/Transforms/Scalar/LoopLoadElimination.cpp:462]:
```suggestion
    auto &DL = Cand.Load->getDataLayout();
```

File: llvm/lib/Transforms/Utils/LoopSimplify.cpp [hunk 5]
```diff
@@ -172,7 +172,7 @@ static void addBlockAndPredsToSet(BasicBlock *InputBB, BasicBlock *StopBlock,
 /// us how to partition the loops.
 static PHINode *findPHIToPartitionLoops(Loop *L, DominatorTree *DT,
                                         AssumptionCache *AC) {
-  const DataLayout &DL = L->getHeader()->getModule()->getDataLayout();
+  const DataLayout &DL = L->getHeader()->getDataLayout();
```
Comment (dtcxzyw) [inline llvm/lib/Transforms/Utils/LoopSimplify.cpp:175]: Do we need an `Loop::getDataLayout`?

**Review by dtcxzyw — APPROVED** [review]
LGTM. Thank you!

**Review by fhahn — APPROVED** [review]
LGTM, this is a nice convenience helper!

**Pull request merged**